ModuleFromCriterion.lua

```diff
@@ -26,10 +26,3 @@ function ModuleFromCriterion:updateGradInput(input, gradOutput)
    self.gradInput[2]:resizeAs(target):zero()
    return self.gradInput
 end
-
-function ModuleFromCriterion:type(t)
-   self.criterion:type(t)
-   self.gradInput[1] = self.gradInput[1]:type(t)
-   self.gradInput[2] = self.gradInput[2]:type(t)
-   return parent.type(self,t)
-end
```

Comment:
`nn.Module:type()` has been modified recently; if you look at

https://github.com/torch/nn/blob/master/Module.lua#L116-L123
https://github.com/torch/nn/blob/master/utils.lua#L3-L15

you'll see that it will call `type` on all members that are of type `nn.Module` or `nn.Criterion`.